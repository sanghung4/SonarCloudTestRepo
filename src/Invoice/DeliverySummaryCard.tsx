import React, { Fragment, useMemo } from 'react';
import { kebabCase } from 'lodash';
import { omit } from 'lodash-es';

import {
  Box,
  Card,
  Grid,
  Skeleton,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { Invoice } from 'generated/graphql';
import { checkStatus } from 'utils/statusMapping';
import { handleAddress } from './util';

type Props = {
  loading: boolean;
  invoice?: Invoice;
};

function DeliverySummaryCard(props: Props) {
  /**
   * Custom hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Memo
   */
  const summaryValues = useMemo(summaryValuesMemo, [props, t]);

  return (
    <Card sx={{ p: isSmallScreen ? 3 : 4, height: 1 }}>
      <Grid container spacing={4}>
        <Grid item md={8}>
          <Box pb={3}>
            <Typography color="primary" variant="h5">
              {t('common.deliverySummary')}
            </Typography>
          </Box>
          <Grid container spacing={4}>
            {summaryValues.map((v) => (
              <Fragment key={v.key}>
                <Grid item xs={5}>
                  <Typography fontWeight={500} whiteSpace="nowrap">
                    {v.key}
                  </Typography>
                </Grid>
                <Grid item xs={7}>
                  {props.loading ? (
                    <Typography>
                      <Skeleton />
                    </Typography>
                  ) : (
                    <Typography
                      data-testid={`delivery-summary-${kebabCase(v.key)}`}
                    >
                      {v.value}
                    </Typography>
                  )}
                </Grid>
              </Fragment>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </Card>
  );

  function summaryValuesMemo() {
    const address = omit(props?.invoice?.address, '__typename');

    return [
      {
        key: t('invoice.shippedTo'),
        value: handleAddress(address)
      },
      {
        key: t('common.shipDate'),
        value: props?.invoice?.shipDate ?? '-'
      },
      {
        key: t('common.deliveryMethod'),
        value: props?.invoice?.deliveryMethod
          ? checkStatus(props.invoice.deliveryMethod)
            ? t(`common.${checkStatus(props.invoice.deliveryMethod)}`)
            : t('common.delivery')
          : '-'
      }
    ];
  }
}

export default DeliverySummaryCard;
