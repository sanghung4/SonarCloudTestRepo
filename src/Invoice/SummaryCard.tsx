import React from 'react';

import {
  Box,
  Card,
  Grid,
  Divider,
  Paper,
  Skeleton,
  Typography,
  alpha
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { format } from 'utils/currency';
import { Invoice } from 'generated/graphql';

type Props = {
  loading?: boolean;
  invoice?: Invoice;
};

function SummaryCard(props: Props) {
  /**
   * Custom hooks
   */
  const { t } = useTranslation();

  return (
    <Card sx={{ px: 2, py: 3 }}>
      <Box pb={3} px={3}>
        <Typography
          color="primary"
          variant="h5"
          data-testid="invoice-summary-header"
        >
          {t('invoice.invoiceSummary')}
        </Typography>
      </Box>
      <Paper
        elevation={0}
        sx={{
          px: 2,
          py: 3,
          bgcolor: (theme) => alpha(theme.palette.primary02.main, 0.05)
        }}
      >
        {/* INVOICE SUBTOTAL*/}

        <Grid container px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.subtotal')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-subtotal"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.subtotal ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary">{t('common.tax')}</Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-tax"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.tax ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary">{t('invoice.other')}</Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-other"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.otherCharges ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
        <Box my={2}>
          <Divider />
        </Box>

        {/*TOTAL AMOUNT AND PAID TO DATE*/}

        <Grid container spacing={2} px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.totalAmountDue')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-totalAmountDue"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.originalAmt ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary" sx={{ fontWeight: 500 }}>
              {t('invoice.paidToDate')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-paidToDate"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.paidToDate ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
        <Box my={2}>
          <Divider />
        </Box>

        {/**REMAINING AMOUNT */}

        <Grid container spacing={2} px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.remainingAmount')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              display="flex"
              justifyContent="flex-end"
              data-testid="invoice-summary-remainingAmount"
            >
              {props.loading ? (
                <Skeleton width={80} />
              ) : (
                format(parseFloat(props?.invoice?.openBalance ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
      </Paper>
    </Card>
  );
}

export default SummaryCard;
