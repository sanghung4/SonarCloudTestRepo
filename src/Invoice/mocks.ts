import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
