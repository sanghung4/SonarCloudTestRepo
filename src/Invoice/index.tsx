import {
  Box,
  Container,
  Grid,
  Skeleton,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { camelCase } from 'lodash-es';
import { useTranslation } from 'react-i18next';
import { useLocation, useParams } from 'react-router-dom';

import Breadcrumbs from 'common/Breadcrumbs';
import Loader from 'common/Loader';
import { ErpSystemEnum, useGetInvoiceQuery } from 'generated/graphql';
// import { PrintIcon } from 'icons';
import DeliverySummaryCard from 'Invoice/DeliverySummaryCard';
import InfoCard from 'Invoice/InfoCard';
import ProductsCard from 'Invoice/ProductsCard';
import SummaryCard from 'Invoice/SummaryCard';
import useDocumentTitle from 'hooks/useDocumentTitle';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

/**
 * Types
 */
type RouterState = {
  fromInvoices?: boolean;
  search?: string;
};

function Invoice() {
  /**
   * Custom hooks
   */
  const { t } = useTranslation();
  const { isSmallScreen } = useScreenSize();
  const { id } = useParams<{ id: string }>();
  const location = useLocation<RouterState>();

  /**
   * Context
   */
  const { selectedAccounts } = useSelectedAccountsContext();
  const isMincron = selectedAccounts?.erpSystemName === ErpSystemEnum.Mincron;

  /**
   * Data
   */
  const { data: invoiceQuery, loading } = useGetInvoiceQuery({
    variables: {
      accountId: selectedAccounts?.billTo?.erpAccountId ?? '',
      invoiceNumber: id
    }
  });

  /**
   * Page Title
   */
  useDocumentTitle(
    t('dynamicPageTitles.invoice', {
      invoiceId: invoiceQuery?.invoice.invoiceNumber ?? ''
    })
  );

  /**
   * Callbacks
   */
  // Commenting this out until we starting fixing printing.
  // const handlePrint = () => window.print();

  return (
    <>
      {loading && <Loader backdrop />}
      <Breadcrumbs
        pageTitle={id}
        config={[
          {
            text: t('common.invoices'),
            to: `/invoices${location.state?.search ?? ''}`
          }
        ]}
      />
      <Container>
        <Grid
          container
          alignItems="center"
          spacing={2}
          pt={isSmallScreen ? 3 : 0}
          pb={3}
        >
          <Grid item md={3}>
            <Typography component="h1" variant="h5">
              {t('invoices.invoiceNumber')}:{' '}
              <Box
                component="span"
                fontWeight={400}
                data-testid="invoice-number"
              >
                {id}
              </Box>
            </Typography>
          </Grid>
          <Grid item md={2} xs={12}>
            <Typography component="h2" variant="h5">
              {t('common.status')}:{' '}
              <Box
                color={
                  invoiceQuery?.invoice?.status === 'Open'
                    ? 'success.main'
                    : 'primary'
                }
                data-testid="invoice-status"
                component="span"
                fontWeight={400}
              >
                {loading ? (
                  <Skeleton
                    width={120}
                    height={24}
                    sx={{ display: 'inline-block', pl: 1 }}
                  />
                ) : (
                  invoiceQuery?.invoice?.status &&
                  t(`orders.${camelCase(invoiceQuery?.invoice?.status ?? '')}`)
                )}
              </Box>
            </Typography>
          </Grid>
          <Grid item md={5.95}>
            <Typography component="h2" variant="h5">
              {t('invoice.terms')}:{' '}
              <Box
                component="span"
                fontWeight={400}
                data-testid="invoice-terms"
              >
                {invoiceQuery?.invoice?.terms}
              </Box>
            </Typography>
          </Grid>
          {/* <Hidden mdDown>
            <Grid item>
              <Button
                startIcon={<PrintIcon color="primary" />}
                onClick={handlePrint}
                variant="inline"
                data-testid="print-button"
                className="noprint"
              >
                {t('common.print')}
              </Button>
            </Grid>
          </Hidden> */}
        </Grid>
        <Grid container spacing={2} pb={3} sx={{ breakInside: 'avoid' }}>
          <Grid item md={6}>
            <InfoCard loading={loading} invoice={invoiceQuery?.invoice} />
          </Grid>
          <Grid item md={6}>
            <DeliverySummaryCard
              loading={loading}
              invoice={invoiceQuery?.invoice}
            />
          </Grid>
          <Grid
            container
            item
            spacing={2}
            direction={isSmallScreen ? 'column-reverse' : 'row'}
          >
            <Grid item md={8}>
              <ProductsCard
                loading={loading}
                invoice={invoiceQuery?.invoice}
                isMincron={isMincron}
              />
            </Grid>
            <Grid item md={4}>
              <SummaryCard loading={loading} invoice={invoiceQuery?.invoice} />
            </Grid>
          </Grid>
        </Grid>
      </Container>
    </>
  );
}

export default Invoice;
