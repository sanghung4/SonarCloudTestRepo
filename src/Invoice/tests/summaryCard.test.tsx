import { render } from 'test-utils/TestWrapper';

import SummaryCard from 'Invoice/SummaryCard';
import { successMincron } from 'Invoice/mocks';

jest.mock('@dialexa/reece-component-library', () => {
  const originalModule = jest.requireActual('@dialexa/reece-component-library');
  return {
    __esModule: true,
    ...originalModule,
    useSnackbar: () => ({
      pushAlert: jest.fn()
    })
  };
});

describe('Invoice - SummaryCard', () => {
  it('Should match snapshot when loading', () => {
    const { container } = render(<SummaryCard loading />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with undefined invoice data', () => {
    const { container } = render(<SummaryCard />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with an invoice', () => {
    const { container } = render(
      <SummaryCard invoice={successMincron.result.data.invoice} />
    );
    expect(container).toMatchSnapshot();
  });
});
