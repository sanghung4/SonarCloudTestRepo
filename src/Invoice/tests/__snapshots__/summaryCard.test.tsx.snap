// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice - SummaryCard Should match snapshot when loading 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
        data-testid="invoice-summary-header"
      >
        Invoice Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Invoice Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-subtotal"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-tax"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Other
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-other"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Total Amount Due
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-totalAmountDue"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Paid to Date
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-paidToDate"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Remaining Amount
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="invoice-summary-remainingAmount"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 80px;"
            />
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Invoice - SummaryCard Should match snapshot with an invoice 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
        data-testid="invoice-summary-header"
      >
        Invoice Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Invoice Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-subtotal"
          >
            $3,184.28
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-tax"
          >
            $262.76
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Other
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-other"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Total Amount Due
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-totalAmountDue"
          >
            $NaN
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Paid to Date
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-paidToDate"
          >
            $1,000.00
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Remaining Amount
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="invoice-summary-remainingAmount"
          >
            $NaN
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Invoice - SummaryCard Should match snapshot with undefined invoice data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
        data-testid="invoice-summary-header"
      >
        Invoice Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Invoice Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-subtotal"
          >
            $0.00
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-tax"
          >
            $0.00
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Other
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-other"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Total Amount Due
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-totalAmountDue"
          >
            $0.00
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Paid to Date
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="invoice-summary-paidToDate"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Remaining Amount
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="invoice-summary-remainingAmount"
          >
            $0.00
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
